test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,326 @@
+
+    assert output.dtype == input.dtype
+    assert output.device == input.device
+
+    # TODO: we can improve performance here by not computing the regular output of the kernel over and over
+    if check_cuda_vs_cpu:
+        _check_kernel_cuda_vs_cpu(kernel, _to_tolerances(check_cuda_vs_cpu), input, *args, **kwargs)
+
+    if check_scripted_vs_eager:
+        _check_kernel_scripted_vs_eager(kernel, _to_tolerances(check_scripted_vs_eager), input, *args, **kwargs)
+
+    if check_batched_vs_single:
+        _check_kernel_batched_vs_single(kernel, _to_tolerances(check_batched_vs_single), input, *args, **kwargs)
+
+
+def _check_kernel_scripted_smoke(dispatcher, input, *args, **kwargs):
+    dispatcher_scripted = _script(dispatcher)
+    dispatcher_scripted(input.as_subclass(torch.Tensor), *args, **kwargs)
+
+
+def _check_dispatcher_dispatch_simple_tensor(dispatcher, kernel, input, *args, **kwargs):
+    if not isinstance(input, datapoints.Image):
+        return
+
+    with mock.patch(f"{dispatcher.__module__}.{kernel.__name__}", wraps=kernel) as spy:
+        output = dispatcher(input.as_subclass(torch.Tensor), *args, **kwargs)
+
+        spy.assert_called_once()
+
+    # We cannot use `isinstance` here since all datapoints are instances of `torch.Tensor` as well
+    assert type(output) is torch.Tensor
+
+
+def _check_dispatcher_dispatch_datapoint(dispatcher, kernel, input, *args, **kwargs):
+    # Due to our complex dispatch architecture for datapoints, we cannot spy on the kernel directly,
+    # but rather have to patch the `Datapoint.__F` attribute to contain the spied on kernel.
+    spy = mock.MagicMock(wraps=kernel)
+    with mock.patch.object(F, kernel.__name__, spy):
+        # Due to Python's name mangling, the `Datapoint.__F` attribute is only accessible from inside the class.
+        # Since that is not the case here, we need to prefix f"_{cls.__name__}"
+        # See https://docs.python.org/3/tutorial/classes.html#private-variables for details
+        with mock.patch.object(datapoints._datapoint.Datapoint, "_Datapoint__F", new=F):
+            output = dispatcher(input, *args, **kwargs)
+
+    spy.assert_called_once()
+    assert isinstance(output, type(input))
+
+    if isinstance(input, datapoints.BoundingBox):
+        assert output.format == input.format
+
+
+def _check_dispatcher_dispatch_pil(dispatcher, input, *args, **kwargs):
+    if not (isinstance(input, datapoints.Image) and input.dtype is torch.uint8):
+        return
+
+    kernel = getattr(F, f"{dispatcher.__name__}_image_pil")
+
+    with mock.patch(f"{dispatcher.__module__}.{kernel.__name__}", wraps=kernel) as spy:
+        output = dispatcher(F.to_image_pil(input), *args, **kwargs)
+
+        spy.assert_called_once()
+
+    assert isinstance(output, PIL.Image.Image)
+
+
+def check_dispatcher(
+    dispatcher,
+    kernel,
+    input,
+    *args,
+    check_dispatch_simple_tensor=True,
+    check_dispatch_datapoint=True,
+    check_dispatch_pil=True,
```

Comment:
Are we ever going to set any of these to False? (Same Q for check_kernel BTW)

Reply:
There are a few instances in v2 tests that deactivate some tests. Thus, I've kept the option for now. Happy to revisit if have significant coverage here. Maybe we can just special case a few ops rather than having this flag.